src/agoradatatools/process.py

```diff
@@ -165,31 +165,27 @@ def process_all_files(config_path: str = None, syn=None):
         )
 
 
-def build_parser():
-    """Builds the argument parser and returns the result.
+app = Typer()
 
-    Returns:
-        argparse.ArgumentParser: argument parser for agora data processing
-    """
-    parser = argparse.ArgumentParser(description="Agora data processing")
-    parser.add_argument(
-        "configpath",
-        help="Agora processing yaml configuration",
-    )
-    parser.add_argument(
-        "-a",
-        "--authtoken",
-        help="Synapse PAT",
-    )
-    return parser
+
+input_path_arg = Argument(..., help="Path to configuration file for processing run")
+synapse_auth_opt = Option(
+    os.environ.get("SYNAPSE_AUTH_TOKEN"),
+    "--auth-token",
+    "-at",
+    help="Synapse authentication token. Defaults to environment variabe $SYNAPSE_AUTH_TOKEN",
+    show_default=False,
+)
 
 
-def main():
-    parser = build_parser()
-    args = parser.parse_args()
-    syn = utils._login_to_synapse(token=args.authtoken)
-    process_all_files(config_path=args.configpath, syn=syn)
+@app.command()
+def process(
```

Comment:
Is there an obvious second command to add? `typer` has a quirk where when there is only one command, it can be called by default only, e.g. `adt <config>`. Which makes the command less explicit. I would prefer `adt process <config>`. I don't think this is a problem if we will likely never have an important second command, but changing the command after people get used to it wouldn't be ideal.

Reply:
Hmm - this would be a good discussion point - I can't think of any off the top of my head, but maybe others have ideas.

Reply:
Thanks @JessterB , we will keep that in mind as we add flags!

Reply:
Another thing to consider -- if we want to use this tool to support Magora ETL in addition to Agora ETL (and we do!), we don't want the command to default to using specific config files. We want to be able to specify e.g.  `test_config.yml` vs `magora_test_config.yml`, or `config.yml` vs `magora_config.yml`.

Reply:
What about breaking up processing and upload into separate commands?
* adt <config> -> triggers processing and synapse upload
* adt process <config> -> triggers processing to local output folder only
* adt upload <config> -> triggers synapse upload from local output folder only

Reply:
This isn't a bad ideal - lets create a ticket to track this work.   Not sure if the above is possible.

Reply:
I created https://sagebionetworks.jira.com/browse/AG-1053 for tracking the separate processing/upload subcommands.

Reply:
@JessterB The current command does not default to any specific config file - any config file that works with the data processing pipeline should work. So `adt magora_test_config.yml` should be supported as is.